br/pkg/lightning/restore/meta_manager.go

```diff
@@ -1031,6 +1085,10 @@ func (m noopTableMetaMgr) InitTableMeta(ctx context.Context) error {
 	return nil
 }
 
+func (m noopTableMetaMgr) ReAllocTableRowIDs(ctx context.Context, _ int64) (int64, int64, error) {
+	return 0, 0, nil
```

Comment:
It seems this shouldn't be a noop implemention.

Reply:
As I concern, `noopTableMetaMgr` is used as long as Lightning is not `local + incremental`. In this sense, Lightning merely allocates rowID from its own memory and needn't reconcile with other instances. So, for `noopTableMetaMgr`, simply returning 0 is trivial and feasible.

What did you suggest to be here?

Reply:
> allocates rowID from its own memory

We can do this in this method, so that we don't care about the actual implementation on the caller side.

Reply:
> We can do this in this method, so that we don't care about the actual implementation on the caller side.

It's hard to do so because `noopTableMetaMgr` is somehow a "placeholder" having no idea about the context of the table restoring procedure. If we utilize this contextless structure, we have to:
1. let `singleMgrBuilder` be no longer contextless
2. maintain restoring information in `noopTableMetaMgr` (for which, this struct will not be "noop" anymore)

It's good to have a global metadata manager though, I think we should be more circumspect about it and do this change in another PR (to make thing clearer). Maybe I just leave a TODO and file an issue now?

Reply:
Yes, it's needed.